### PR TITLE
feature/APPEALS-33575

### PR DESCRIPTION
**Pull request by mikefinneran**
IRSR 
[APPEALS-33576](https://jira.devops.va.gov/browse/APPEALS-33576) Refine Existing Metrics
[APPEALS-33577](https://jira.devops.va.gov/browse/APPEALS-33577) Implement New Metrics

# Description
Refine existing metrics and implement new metrics for Reader testing.  
The reader metrics are being used with a select user group to capture the latency time of various Reader actions so that they can be used for analysis.

## Acceptance Criteria
- [x ] Code compiles correctly

## Testing Plan
 https://jira.devops.va.gov/browse/APPEALS-33575

 BEFORE|AFTER
 ---|---

# Backend
## Database Changes
*Only for Schema Changes*

* [x ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ x] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ x] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

# Best practices
## Code Documentation Updates
- [ x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x ] No new code climate issues added

### Monitoring
- [ x] Are performance metrics (e.g., response time, throughput) being tracked?
- [ x] Are key application components monitored (e.g., database, cache, queues)?
- [ x] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [x ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ x] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ x] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ x] Is log retention and rotation configured correctly?
- [ x] Are logs being forwarded to a centralized logging system if needed?

### Error Handling
- [x ] Are errors being caught and handled gracefully?
- [ x] Are appropriate error messages being displayed to users?
- [ x] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ x] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ x] Are custom exceptions defined and used where appropriate?
- [ x] Is exception handling consistent throughout the codebase?
- [ x] Are exceptions logged with relevant context and stack trace information?
- [ x] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

**Review by nkutub — APPROVED** [review]

**Review by zurbergram — APPROVED** [review]

File: client/app/util/ApiUtil.js [hunk 1]
```diff
@@ -40,23 +42,127 @@ export const getHeadersObject = (options = {}) => {
...
+const errorHandling = (url, error, method, options = {}) => {
+  const id = uuid.v4();
+  const message = `UUID: ${id}.\nProblem with ${method} ${url}.\n${error}`;
+
+  console.error(new Error(message));
+  options.t1 = performance.now();
+  options.end = moment().format();
+  options.duration = options.t1 - options.t0;
+
+  // Need to renable this check before going to master
```
Comment (cacevesva) [inline client/app/util/ApiUtil.js:65]: This needs to be verified.

Comment (mattray-gov) [reply client/app/util/ApiUtil.js:65]: This comment was not deleted when the feature toggle was re-enabled.

File: client/app/util/ApiUtil.js [hunk 2]
```diff
@@ -40,23 +42,127 @@ export const getHeadersObject = (options = {}) => {
...
+
+    postMetricLogs(data);
+  }
+};
+
+const successHandling = (url, res, method, options = {}) => {
+  const id = uuid.v4();
+  const message = `UUID: ${id}.\nSuccess with ${method} ${url}.\n${res.status}`;
+
+  // Need to renable this check before going to master
```
[the remark this answers is not in the document]
Comment (mattray-gov) [reply client/app/util/ApiUtil.js:95]: This comment was not deleted when the feature toggle was re-enabled.

[the remark this answers is not in the document]
Comment (mikefinneran) [reply client/app/util/ApiUtil.js:95]: Follow up PR to remove the comments will go to production on 11/22 with a CR to adjust the feature toggles for the test group
https://github.com/department-of-veterans-affairs/caseflow/pull/19950

File: client/app/util/ApiUtil.js (hunk 1 above)
Comment (mikefinneran) [reply client/app/util/ApiUtil.js:65]: Follow up PR to remove the comments will go to production on 11/22 with a CR to adjust the feature toggles for the test group
https://github.com/department-of-veterans-affairs/caseflow/pull/19950

**Pull request merged**